craft_application/application.py

```diff
@@ -473,6 +474,9 @@ def run_managed(self, platform: str | None, build_for: str | None) -> None:
         if self._enable_fetch_service:
             self.services.fetch.shutdown(force=True)
 
+        if self._enable_testing:
+            self._run_tests()
```

Comment:
At some point maybe we should have a command hook to run things in the outer instance.

Reply:
We keep having cases where this would be useful and we keep making workarounds like this.

Reply:
I think we're at the limit of the current design's capabilities really. For example, what about destructive mode?